src/ddmd/traits.d

```diff
@@ -942,6 +942,13 @@ extern (C++) Expression semanticTraits(TraitsExp e, Scope* sc)
         {
             if (!sm)
                 return 1;
+            if (auto decl = sm.isDeclaration())
```

Comment:
__traits(allMembers, ...) shouldn't pick up loop variables of static foreach loops in member scope.